google_test/JsonParser_test.cpp

```diff
@@ -1,7 +1,111 @@
 #include "../JsonParser.h"
+#include "../creature.h"
 #include <iostream>
 #include <gtest/gtest.h>
 
+TEST(CreatureTests, FightEndTest)
+{
+    Creature* h1 = Creature::parseUnit("../Units/Kakarott.json");
+    Creature* h2 = Creature::parseUnit("../Units/Maple.json");
+    h1->fight(h2);
+    ASSERT_TRUE(h1->isDead() || h2->isDead());
+}
+
+TEST(CreatureTests, FightResult)
+{
+    Creature* h1 = Creature::parseUnit("../Units/Maple.json");
+    Creature* h2 = Creature::parseUnit("../Units/Sally.json");
+    h1->fight(h2);
+    ASSERT_DOUBLE_EQ(h1->getLife(),-3);
+    ASSERT_DOUBLE_EQ(h2->getLife(),485);
+}
```

Comment:
Ez a két teszt is nagy részben átfedi egymást. Szükséges így az első?

Reply:
Szükséges, mert a FightEndTest-ben az isDead()-t vizsgáljuk, hogy a csata kimenete jó-e, tehát van-e halott, míg a FightResult-ban pedig a harcosok csata utáni elvárt életét vizsgáljuk a getLife()-al.